### PR TITLE
Fix comptime_int bitwise operators (#1387)

### LINKED ISSUE #1387: comptime_int bitwise operators with negative values give incorrect results
This is with the stage1 compiler.
```zig
std.debug.warn("{}\n", i64(3 & -1)); // Prints 1, but should be 3
std.debug.warn("{}\n", i64(3 | -1)); // Prints 140338432934211, but should be -1
std.debug.warn("{}\n", i64(3 ^ -1)); // Prints 140678271664562, but should be -4
```

### PR DESCRIPTION
**Pull request by kristate**
Fixes #1387 

**#1387**
* [X] `i64(3 & -1) == 3;`
* [X] `i64(3 & 1) == 1;`
* [X] `i64(-3 & -1) == -3;`
* [X] `i64(3 | -1) == -1;`
* [X] `i64(3 ^ -1) == -4;`

### CODE REVIEW

File: src/bigint.cpp [hunk 1]
```diff
@@ -437,6 +435,21 @@ bool mul_u64_overflow(uint64_t op1, uint64_t op2, uint64_t *result) {
 }
 #endif
 
+static inline void _twos_complement_inplace(BigInt *bn) {
```
Comment (andrewrk) [inline src/bigint.cpp]: No explicit `inline` unless:
 * you need it so that you can put the function in a header file
 * you have a benchmark to demonstrate that it's faster and not too much code size increase
 * the function semantics depends on always being inlined. in this case you would need a compiler-specific attribute anyway.

Also, no need for leading underscore on static functions.

File: src/bigint.cpp [hunk 2]
```diff
@@ -1447,9 +1410,12 @@ void bigint_negate_wrap(BigInt *dest, const BigInt *op, size_t bit_count) {
 }
 
 void bigint_not(BigInt *dest, const BigInt *op, size_t bit_count, bool is_signed) {
-    if (bit_count == 0) {
-        bigint_init_unsigned(dest, 0);
-        return;
+    if (!bit_count) {
+        bit_count = (op->digit_count * 64) - bigint_clz(op, op->digit_count * 64) + (is_signed ? 1 : 0);
```
Comment (andrewrk) [inline src/bigint.cpp]: why this change?

this function is supposed to evaluate binary not based on the bit count passed in as a parameter. So if the passed in value is 0 then binary not of 0 is 0.

Comment (kristate) [reply src/bigint.cpp]: @andrewrk I will document this function.

> So if the passed in value is 0 then binary not of 0 is 0.

I think that is an incorrect assumption -- `bit_count` only specifies how many bits should be included in the `not` operation; furthermore, `BigInt` is architected only to store a value. zig seems to have a u0 type, but I don't see how this is valuable or used.

This implementation correctly returns `not` of `op` into `dest` with optional bound of `bit_count`.

File: src/bigint.cpp (hunk 1 above)
Comment (kristate) [reply src/bigint.cpp]: Okay, I will change the signature.

**Comment by andrewrk** [comment]
> `i64(-3 & -1) == -3`

This test case is incorrect.
